azurelinuxagent/common/utils/timeutil.py

```diff
@@ -1,39 +1,20 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the Apache License.
-import datetime
-
-
-def create_timestamp(dt=None):
-    """
-    Returns a string with the given datetime in iso format. If no datetime is given as parameter, it
-    uses datetime.utcnow().
-    """
-    if dt is None:
-        dt = datetime.datetime.utcnow()
-    return dt.isoformat()
-
-
-def create_history_timestamp(dt=None):
-    """
-    Returns a string with the given datetime formatted as a timestamp for the agent's history folder
-    """
-    if dt is None:
-        dt = datetime.datetime.utcnow()
-    return dt.strftime('%Y-%m-%dT%H-%M-%S')
 
+import datetime
 
-def datetime_to_ticks(dt):
+def create_utc_timestamp(dt):
```

Comment:
We were using different formats for timestamps: one for the history folder, another for logging/events/goal state, and yet another for the state of the last FastTrack goal state.

Timestamps in the goal stare (produced with strftime("%Y-%m-%dT%H:%M:%S.%fZ") are compared with those in FastTrack (produced with datetime.isoformat). There are small differences in these 2 formats (isoformat does not use 'Z', and it does not produce a value for microseconds whey they are 0) that can lead to subtle bugs.

I move the history timestamps to GoalStateHistory for its own internal use, and I changed the rest of the code to use this new function.